Vic2ToHoI4/Source/OutHoi4/OutHoi4World.cpp

```diff
@@ -0,0 +1,549 @@
+		{
+			outputIdeaGraphics(ideasFile, *country.second);
+		}
+	}
+	ideasFile << "\n";
+	ideasFile << "}\n";
+}
+
+
+void HoI4::outputRelations(const std::string& outputName)
+{
+	LOG(LogLevel::Info) << "\t\tWriting opinion modifiers";
+
+	if (!Utils::TryCreateFolder("output/" + outputName + "/common/opinion_modifiers"))
+	{
+		throw std::runtime_error("Could not create output/" + outputName + "/common/opinion_modifiers/");
+	}
+	std::ofstream out("output/" + outputName + "/common/opinion_modifiers/01_opinion_modifiers.txt");
+	if (!out.is_open())
+	{
+		throw std::runtime_error(
+			 "Could not create output/" + outputName + "/common/opinion_modifiers/01_opinion_modifiers.txt");
+	}
+
+	out << "opinion_modifiers = {\n";
+	for (auto i = -200; i <= 200; i++)
+	{
+		if (i < 0)
+		{
+			out << "negative_";
+		}
+		else
+		{
+			out << "positive_";
+		}
+		out << std::abs(i) << " = {\n";
+		out << "\tvalue = " << i << "\n";
+		out << "}\n";
+	}
+	out << "private_channels_trade = {\n";
+	out << "\ttrade = yes\n";
+	out << "\tvalue = 15\n";
+	out << "}\n";
+	out << "absolutist_in_government = {\n";
+	out << "\tvalue = 30\n";
+	out << "}\n";
+	out << "communism_in_government = {\n";
+	out << "\tvalue = 30\n";
+	out << "}\n";
+	out << "fascism_in_government = {\n";
+	out << "\tvalue = 30\n";
+	out << "}\n";
+	out << "radical_in_government = {\n";
+	out << "\tvalue = 30\n";
+	out << "}\n";
+
+	out << "}\n";
+
+	out.close();
+}
+
+
+void HoI4::outputLeaderTraits(const std::map<std::string, std::vector<std::string>>& ideologicalLeaderTraits,
+	 const std::set<std::string>& majorIdeologies,
+	 const std::string& outputName)
+{
+	LOG(LogLevel::Info) << "\t\tWriting leader traits";
+
+	std::ofstream traitsFile("output/" + outputName + "/common/country_leader/converterTraits.txt");
+	if (!traitsFile.is_open())
+	{
+		throw std::runtime_error("Could not create output/" + outputName + "/common/country_leader/converterTraits.txt");
+	}
+
+	traitsFile << "leader_traits = {\n";
+	for (const auto& majorIdeology: majorIdeologies)
+	{
+		if (auto ideologyTraits = ideologicalLeaderTraits.find(majorIdeology);
+			 ideologyTraits != ideologicalLeaderTraits.end())
+		{
+			for (const auto& trait: ideologyTraits->second)
+			{
+				traitsFile << "\t";
+				traitsFile << trait;
+			}
+		}
+	}
+	traitsFile << "}";
+	traitsFile.close();
+}
+
+
+void HoI4::outputBookmarks(const std::vector<std::shared_ptr<Country>>& greatPowers,
+	 const std::map<std::string, std::shared_ptr<Country>>& countries,
+	 const std::string& outputName)
+{
+	LOG(LogLevel::Info) << "\t\tWriting bookmarks";
+
+	std::ofstream bookmarkFile("output/" + outputName + "/common/bookmarks/the_gathering_storm.txt");
+	if (!bookmarkFile.is_open())
+	{
+		throw std::runtime_error("Could not create output/" + outputName + "/common/bookmarks/the_gathering_storm.txt");
+	}
+
+	bookmarkFile << "bookmarks = {\n";
+	bookmarkFile << "\tbookmark = {\n";
+	bookmarkFile << "\t\tname = "
```

Comment:
Either \ts missing here or pluses or something. This looks suspicious.

Reply:
The formatter did something weird, I think. Fixed.